### PR TITLE
docs and some small treats for wdi5 2.0

### PR DESCRIPTION
**Pull request by vobu**
- reworked the documentation 
- add an `esm` usage example for `wdi5`
- syntactic sugar for switching between SAP Build WorkZone's shell and app area
- devx: `jsconfig.json` samples in CJS- and ESM-flavor

### CODE REVIEW

**Comment by vobu** [comment]
Hey @wridgeu and @marianfoo - sorry for the review ambush 🥷  Added you as reviewers in the hopes that some fresh eyes on documentation will find gaps, shortcomings and typos better :)

**Comment by vobu** [comment]
btw: failed test is a flaky one, as the wz shell is ...well... anyway, ignore it please 
https://github.com/ui5-community/wdi5/actions/runs/6051525181/job/16423173682

**Review by wridgeu — APPROVED** [review]
Some really minor things that caught my eye but overall it read nicely and LGTM.

File: examples/ui5-js-app-esm/webapp/controller/App.controller.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+"use strict"
```
[the remark this answers is not in the document]
Comment (vobu) [reply examples/ui5-js-app-esm/webapp/controller/App.controller.js:1]: thanks, rm'ed all occurrences (use strict is so yesterday :) )

File: examples/ui5-js-app-esm/webapp/controller/Main.controller.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                this.getOwnerComponent().getModel().read("/Customers('TRAIH')")
+
+                const jData = {
+                    inputValue: "test Input Value !!!",
+                    buttonText: "Don't press me !!! -> binded",
+                    checkbox: false,
+                    barcode: ""
+                }
+                // TODO:
+                // searchValue: "search Value"
```
Comment (marianfoo) [inline examples/ui5-js-app-esm/webapp/controller/Main.controller.js:20]: TODO will stay?

File: examples/ui5-js-app-esm/webapp/test/e2e/wdio.conf.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+export const config = {
+    wdi5: {
+        logLevel: "verbose"
+    },
+    baseUrl: "http://localhost:8082/index.html",
```
Comment (marianfoo) [inline examples/ui5-js-app-esm/webapp/test/e2e/wdio.conf.js:5]: why is the port 8082? App running at 8008, right?

Comment (marianfoo) [reply examples/ui5-js-app-esm/webapp/test/e2e/wdio.conf.js:5]: ah, i see, you run it with 8082 at github workflow

File: examples/ui5-js-app-esm/package.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  "name": "ui5-app-esm",
+  "version": "0.8.15-notimportant",
+  "private": true,
+  "description": "sample ESM ui5 app for testing wdi5",
+  "license": "UNLICENSED",
+  "author": "j&s-soft GmbH",
+  "main": "webapp/index.html",
+  "type": "module",
+  "scripts": {
+    "start": "ui5 serve",
```
Comment (marianfoo) [inline examples/ui5-js-app-esm/package.json:11]:
```suggestion
    "start": "ui5 serve -p 8082",
```

File: examples/ui5-js-app-esm/webapp/Component.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+sap.ui.define(
+    [
+        "sap/ui/core/UIComponent",
+        "sap/ui/Device",
+        "test/Sample/model/models",
+        "sap/ui/core/ComponentSupport" // make sure to include the ComponentSupport in the bundle
+    ],
+    function (UIComponent, Device, models) {
```
Comment (marianfoo) [inline examples/ui5-js-app-esm/webapp/Component.js:8]:
```suggestion
    function (UIComponent, models) {
```

**Review by marianfoo — COMMENTED** [review]
Just a few suggestions

**Comment by vobu** [comment]
closing this in favor of 2 new ones for better separation of concerns.
thx @marianfoo and @wridgeu for the reviews!

**Pull request closed without merging**